### PR TITLE
Joe security new integration

### PR DESCRIPTION
**Pull request by yaakovi**
fixes: https://github.com/demisto/etc/issues/9621

### CODE REVIEW

**Comment by dorsha** [comment]
Don't forget the documentation.

**Comment by yaakovi** [comment]
Overview:
Joe Security Sandbox detects and analyses potential malicious files.
Using the integration, one can use it to analyse url links, sample files of different formats (see additional information) on different machine types (windows, android, iOS and Mac OS X).


To set up Joe Security on Demisto:
1. Login to: https://jbxcloud.joesecurity.org
2. Go to user settings (press the button at the top right corner and click “settings” in the dropdown menu)
3. Go to “API Key” section
4. Check the “I Agree” checkbox and click the “Generate API key” button.
5. Use the API key to set up the integration in Demisto.

To set up the integration on Demisto:
1. Go to 'Settings > Integrations > Servers & Services'
2. Locate Joe Security by searching for it using the search box on the top of the page.
3. Click 'Add instance' to create and configure a new integration
4. You should configure the following settings:
    * Name: a textual name for the integration instance.
    * API Key: the generated API key from previous section.
    * Do not validate server certificate (insecure)
    * Use system proxy settings: should the integration use Demisto proxy
    * Test: Checks correctness of URLs and token.

Top Use-cases:
To send a sampling file to joe security:
    * Drag-and-drop a file to integration war-room
    * Use the following command:
!joe-analysis-submit-sample sample=<file entry id>
    * Alternatively, you can sample a file from the network using “sample_url” instead of “sample”.

To get information on an old analysis, use:
!joe-analysis-info <analysis ID>

Commands:
1.Command name: joe-is-online
Inputs:
-
Context Output:
-
Raw output:

2.Command name: joe-analysis-submit-url
Inputs:
1. url - sample url
2. should_wait - Should the command poll for the result of the analysis
Context Output:
Path: Joe. Analysis.WebID
Description: Web ID
Raw output:

3.Command name: joe-analysis-info
Inputs:
1. webid - Web ID
Context Output:
Path: Joe. Analysis.WebID
Description: Web ID
Path: Joe. Analysis.FileName
Description: Sample Data
Path: Joe. Analysis.Status
Description: Analysis Status
Path: Joe. Analysis.Comments
Description: Analysis Comments
Path: Joe. Analysis.Time
Description: Submitted Time
Path: Joe. Analysis.Runs
Description: Application used for analysis
Path: Joe. Analysis.Result
Description: Analysis Results
Raw output:

4.Command name: joe-list-analysis
Inputs:
-
Context Output:
Path: Joe. Analysis.WebID
Description: Web ID
Raw output:

5.Command name: joe-analysis-submit-sample
Inputs:
1.sample - Sample File
2.sample_url - Url to a sample file
3.should_wait - Should the command poll for the result of the analysis
Context Output:
Path: Joe. Analysis.WebID
Description: Web ID
Raw output:


Additional information:
Sampling Supported files are: 
- PE (EXE, DLL, SYS, PIF, SCR, CPL etc)
- Office files (DOC(X)(M), PPT(X)(M), XLS(X)(M), DOC(X)(M), PPT(X)(M), XLS(X)(M))
- PDF, HWP, JTD, MSG, EML, JAR, JS, JSE, VBS, VBE, WSF, CRX, XPI, APK, URL

File: Integrations/integration-JoeSecurity.yml [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+
+    ''' HELPER FUNCTIONS '''
+    def http_post(url_suffix, data=None, files=None):
+        data = {} if data is None else data
+
+        data.setdefault('apikey', demisto.params()['api_key'])
+        LOG('running request with url=%s\tdata=%s\tfiles=%s' % (BASE_URL + url_suffix,
+            data, files, ))
+        try:
+            res = requests.post(BASE_URL + url_suffix,
```
Comment (anara123) [inline Integrations/integration-JoeSecurity.yml]: you don't handle error status_code in http_post, but you should.

File: Integrations/integration-JoeSecurity.yml [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+
+    def analyse_url_request(url, should_wait):
+        res = http_post('v2/analysis/submit',
+            data={'accept-tac': 1, 'url' : url})
+
+        if 'errors' in res:
+            LOG('Error! in command analyse_url: url=%s' % (url, ))
+            LOG('got the following errors:\n' + '\n'.join(e['message'] for e in res['errors']))
+            return 'Failure'
```
Comment (anara123) [inline Integrations/integration-JoeSecurity.yml]: throw exception here instead returning 'Failure'

**Pull request merged**